core/src/main/java/org/elasticsearch/gateway/ReplicaShardAllocator.java

```diff
@@ -116,6 +116,11 @@ public boolean allocateUnassigned(RoutingAllocation allocation) {
                 continue;
             }
 
+            // if we are allocating a replica because of index creation, no need to go and find a copy, there isn't one...
+            if (shard.allocatedPostIndexCreate() == false) {
```

Comment:
Ahh okay, this is only when the index is newly created, so there's no point in it for shadow replicas, ignore my previous comment then :)